### DISCUSSION TITLE
会有deepin-cloud-build-ci吗？

### DISCUSSION THREAD

**Discussion by 8MiYile** [linuxdeepin/developer-center#4046] [Q&A | 问答板块] [answered]
我希望deepin能整一个deepin-cloud-build-ci出来，方便GitHub Actions对接的

**Comment by 8MiYile** [comment]
https://github.com/obsproject/obs-studio
想要这家伙能自动编译Deepin用的deb包（）

**Reply by Decodetalkers** [reply]
> https://github.com/obsproject/obs-studio 想要这家伙能自动编译Deepin用的deb包（）

我们现在有obs流啊。只是还没有开放注册

**Reply by 8MiYile** [reply]
那个啥，我寻思商店里面的OBS Studio已经落后官网好几个版本 要不。。。
还是说 之后先从GitHub Actions开放？

**Reply by Decodetalkers** [reply]
> 那个啥，我寻思商店里面的OBS Studio已经落后官网好几个版本 要不。。。
> 还是说 之后先从GitHub Actions开放？

要不你来尝试打包？我们有社区仓库的

**Reply by 8MiYile** [reply]
有类似的CI/CD嘛。。我的编译环境有被破坏过。。。

**Comment by BLumia** [comment]
有为进仓提供的 [Open Build Service](https://build.deepin.com/) （简称也是 OBS）实例来进行打包。向 linuxdeepin 与 deepin-community 组织下的仓库发起贡献的话，默认会使用我们的 OBS 实例进行打包。如果你有意愿将 obs-studio 添加到 deepin v23 起的系统仓库的话，可以申请往 deepin-community 创建仓库，然后打包维护。整个过程都可以在 GitHub 进行，如果有疑问的话也欢迎加入 sysdev 组讨论，可加 [Matrix](https://wiki.deepin.org/Matrix) 群 `#deepin-sig-sysdev-team:matrix.org`。

如果你的意思是希望自己编写 GitHub Action 工作流来构建适用于 deepin 环境的软件包的话，可以考虑用 [docker](https://hub.docker.com/u/linuxdeepin)，或者用 pbuilder 使用[这个 baseimage](https://ci.deepin.com/repo/baseimage/) 进行构建。

**Reply by 8MiYile** [reply]
嗯 我想问一下 如何将OBS Studio仓库 fork进 deepin-community ？
亦或者是说 如何申请

**Reply by BLumia** [reply, marked as answer]
可以参见 [deepin-community协作指南](https://wiki.deepin.org/deepin-community%E5%8D%8F%E4%BD%9C%E6%8C%87%E5%8D%97) 这个页面的说明。大致上的话是这样：

1. 向 [deepin-community/SIG](https://github.com/deepin-community/SIG) 仓库发起 PR，编辑 repos.yml 配置文件来追加仓库
2. PR 合入后会根据所编辑的配置自动创建仓库
3. Fork 所创建的仓库，向你 Fork 的仓库中进行提交，然后发起 PR
4. PR 合入后即完成一次更新，此时软件包即位于 Unstable（或者叫 commit）仓库。

这时候可以认为 v23 源已经有这个包了，但若需要最终在版本发布时能在稳定仓库中也存在这个包，就只剩下集成了。集成的步骤大概是这样：

1. 向 [deepin-community/Repository-Integration](https://github.com/deepin-community/Repository-Integration/) 仓库发起 PR，编辑 intergration.yml
2. PR 创建后，待 OBS 自动构建完成，会自动在 linuxdeepin/developer-center 的 Issue 面板创建对应的集成 Issue
3. 在 Issue 中，主动补全一些测试建议信息，描述哪些内容需要被测试人员关注
4. 顺利完成测试后，仓库会被集成到 testing 仓库，最终系统版本发布时，会进入 stable 仓库（参见 [仓库流转规范](https://wiki.deepin.org/%E4%BB%93%E5%BA%93%E6%B5%81%E8%BD%AC%E8%A7%84%E8%8C%83)）

p.s. 如果有意向维护+首次作这件事的话，还是建议能在 Matrix 群一起交流的，这样遇到问题的反馈也会更及时一些 :)
